code/datums/components/shrink.dm

```diff
@@ -25,6 +26,13 @@
 			if(ishuman(C))
 				var/mob/living/carbon/human/H = C
 				H.physiology.damage_resistance -= 100//carbons take double damage while shrunk
+		if(!L.GetComponent(/datum/component/squashable))
+			newsquash = L.AddComponent( \
+				/datum/component/squashable, \
+				squash_chance = 75, \
+				squash_damage = 10, \
+				squash_flags = SQUASHED_ALWAYS_IF_DEAD|SQUASHED_DONT_SQUASH_IN_CONTENTS, \
+			)
```

Comment:
Using `GetComponent` is generally bad, and components should handle what happens if a datum already has the passed component:
```suggestion
		newsquash = L.AddComponent( \
			/datum/component/squashable, \
			squash_chance = 75, \
			squash_damage = 10, \
			squash_flags = SQUASHED_ALWAYS_IF_DEAD|SQUASHED_DONT_SQUASH_IN_CONTENTS, \
		)
```
Note, `/datum/component/squashable` is `dupe_mode = COMPONENT_DUPE_HIGHLANDER`. 
Meaning it will replace existing ones. Which is probably fine? 
If you want to avoid doing that, you would need to change its dupe mode.

Reply:
Yeah I wanted to avoid doing, I guess I could replace the dupe mode but I just wasn't sure if anything else relied on it

Reply:
Actually wouldn't this introduce a potential issue where, if the mob was previously squishable like a roach, they wouldn't be after unshrinking?

Reply:
Eh that's a good point, I forgot that `AddComponent` returns existing instances in the event of a dupe mode conflict. Fairly certain this is still solvable with dupe modes in some way but I'm not sure, so leave it as it I suppose